client/state/signup/verticals/reducer.js

```diff
@@ -7,8 +7,15 @@ import { createReducer } from 'state/utils';
 import { SIGNUP_VERTICALS_SET } from 'state/action-types';
 
 export default createReducer( null, {
-	[ SIGNUP_VERTICALS_SET ]: ( state, action ) => ( {
-		...state,
-		[ action.search.trim().toLowerCase() ]: action.verticals,
-	} ),
+	[ SIGNUP_VERTICALS_SET ]: ( state, action ) => {
```

Comment:
I'm not especially familiar with the rest of this PR, but this bit jumped out at me. We tend to use `createReducer` without thinking about it where it adds little or no value and I'd discourage that as unnecessary overhead. Built in control flow like `switch` and `if` are almost always preferable.

It looks like a great candidate for `keyedReducer`, however:

```js
export default keyedReducer( 'siteTypeId', ( state = {}, action ) => {
  if ( action.type === SIGNUP_VERTICALS_SET ) {
    return {
      ...state,
      [ action.search.trim().toLowerCase() ]: action.verticals,
    };
  }
  return state;
} );
```

Reply:
Good call! Since this PR does not really add `createReducer`, only reworks an existing call, I won't add this change here. Instead, I added an issue so we can track it and address during a janitorial phase:

https://github.com/Automattic/zelda-private/issues/33